Source/p_enemy.c

```diff
@@ -2905,7 +2905,8 @@ void A_MonsterMeleeAttack(mobj_t *actor)
   if (!P_CheckRange(actor, range))
     return;
 
-  S_StartSound(actor, hitsound);
+  if (hitsound > 0)
+    S_StartSound(actor, hitsound);
```

Comment:
We should probably just ignore `sfx_id == sfx_None` in `S_StartSound()`.

https://github.com/coelckers/prboom-plus/commit/06908671cb84e6b90d112c665b496fb493e21cc2